# Ensure integer size of exported images. See https://github.com/pyqtgraph/pyqtgraph/pull/356
# authors: D.Penko, S.D.Pinches, 2019
diff -Nru pyqtgraph-0.10.0-orig/pyqtgraph/exporters/ImageExporter.py pyqtgraph-0.10.0/pyqtgraph/exporters/ImageExporter.py
--- pyqtgraph-0.10.0-orig/pyqtgraph/exporters/ImageExporter.py	2016-10-17 18:22:41.000000000 +0200
+++ pyqtgraph-0.10.0/pyqtgraph/exporters/ImageExporter.py	2020-06-24 10:50:36.544574000 +0200
@@ -64,10 +64,11 @@
         
         #self.png = QtGui.QImage(targetRect.size(), QtGui.QImage.Format_ARGB32)
         #self.png.fill(pyqtgraph.mkColor(self.params['background']))
-        w, h = self.params['width'], self.params['height']
+        w = int(self.params['width'])
+        h = int(self.params['height'])
         if w == 0 or h == 0:
             raise Exception("Cannot export image with size=0 (requested export size is %dx%d)" % (w,h))
-        bg = np.empty((self.params['width'], self.params['height'], 4), dtype=np.ubyte)
+        bg = np.empty((h, w, 4), dtype=np.ubyte)
         color = self.params['background']
         bg[:,:,0] = color.blue()
         bg[:,:,1] = color.green()
